src/modules/75-cd/components/PipelineSteps/DeployInfrastructureStep/AddEditEnvironmentModal.tsx

```diff
@@ -120,8 +120,8 @@ export default function AddEditEnvironmentModal({
           description: defaultTo(description, ''),
           tags: defaultTo(tags, {}),
           type: defaultTo(type, ''),
-          orgIdentifier: data.environment?.orgIdentifier,
-          projectIdentifier: data.environment?.projectIdentifier,
+          orgIdentifier: defaultTo(data.environment?.orgIdentifier, defaultTo(orgIdentifier, '')),
+          projectIdentifier: defaultTo(data.environment?.projectIdentifier, defaultTo(projectIdentifier, '')),
```

Comment:
Just curious, what's the benefit go using `defaultTo` instead of `||`?

```suggestion
          orgIdentifier: data.environment?.orgIdentifier || orgIdentifier || '',
          projectIdentifier: data.environment?.projectIdentifier || projectIdentifier || '',

Reply:
huh weird.. this was not my change. not sure how this crept in. probably during the rebase.. but there were no conflicts 🤔 

but to answer your question, using `defaultTo` doesn't increase the cognitive complexity in Sonar. `||` is considered a condition, which adds to the complexity and also adds a branch to be tested for coverage.

Reply:
Don't ya just love when we have to add extra code and actual complexity to get around a tool we've introduced? The whole FE industry is just messed up. lol